### ISSUE TITLE
TypeError: should_bypass_proxies_patched() missing 1 required positional argument: 'no_proxy'

### ISSUE THREAD

**Issue by tarelli** [conda/conda#7503]
**I'm submitting a...**
  - [X] bug report
  - [ ] feature request

## Current Behavior
Running `RUN conda create --name snakes python=2` inside a Docker container built by Travis throws the error. This has always worked, this failure started happening today ~2 hours ago.

```
Step 22/36 : RUN conda create --name snakes python=2
 ---> Running in beb081e7f2e4
Solving environment: ...working... failed

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/opt/conda/lib/python3.5/site-packages/conda/exceptions.py", line 819, in __call__
        return func(*args, **kwargs)
      File "/opt/conda/lib/python3.5/site-packages/conda/cli/main.py", line 78, in _main
        exit_code = do_call(args, p)
      File "/opt/conda/lib/python3.5/site-packages/conda/cli/conda_argparse.py", line 77, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/opt/conda/lib/python3.5/site-packages/conda/cli/main_create.py", line 11, in execute
        install(args, parser, 'create')
      File "/opt/conda/lib/python3.5/site-packages/conda/cli/install.py", line 235, in install
        force_reinstall=context.force,
      File "/opt/conda/lib/python3.5/site-packages/conda/core/solve.py", line 506, in solve_for_transaction
        force_remove, force_reinstall)
      File "/opt/conda/lib/python3.5/site-packages/conda/core/solve.py", line 439, in solve_for_diff
        final_precs = self.solve_final_state(deps_modifier, prune, ignore_pinned, force_remove)
      File "/opt/conda/lib/python3.5/site-packages/conda/core/solve.py", line 180, in solve_final_state
        index, r = self._prepare(prepared_specs)
      File "/opt/conda/lib/python3.5/site-packages/conda/core/solve.py", line 580, in _prepare
        self.subdirs, prepared_specs)
      File "/opt/conda/lib/python3.5/site-packages/conda/core/index.py", line 215, in get_reduced_index
        new_records = query_all(spec)
      File "/opt/conda/lib/python3.5/site-packages/conda/core/index.py", line 184, in query_all
        return tuple(concat(future.result() for future in as_completed(futures)))
      File "/opt/conda/lib/python3.5/site-packages/conda/core/subdir_data.py", line 94, in query
        self.load()
      File "/opt/conda/lib/python3.5/site-packages/conda/core/subdir_data.py", line 148, in load
        _internal_state = self._load()
      File "/opt/conda/lib/python3.5/site-packages/conda/core/subdir_data.py", line 216, in _load
        mod_etag_headers.get('_mod'))
      File "/opt/conda/lib/python3.5/site-packages/conda/core/subdir_data.py", line 416, in fetch_repodata_remote_request
        timeout=timeout)
      File "/opt/conda/lib/python3.5/site-packages/pip/_vendor/requests/sessions.py", line 487, in get
        return self.request('GET', url, **kwargs)
      File "/opt/conda/lib/python3.5/site-packages/pip/_vendor/requests/sessions.py", line 466, in request
        prep.url, proxies, stream, verify, cert
      File "/opt/conda/lib/python3.5/site-packages/pip/_vendor/requests/sessions.py", line 626, in merge_environment_settings
        env_proxies = get_environ_proxies(url) or {}
      File "/opt/conda/lib/python3.5/site-packages/pip/_vendor/requests/utils.py", line 568, in get_environ_proxies
        if should_bypass_proxies(url):
    TypeError: should_bypass_proxies_patched() missing 1 required positional argument: 'no_proxy'

`$ /opt/conda/bin/conda create --name snakes python=2`

  environment variables:
                 CIO_TEST=<not set>
                CONDA_DIR=/opt/conda
               CONDA_ROOT=/opt/conda
                     PATH=/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin
                          :/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : None
       user config file : /home/jovyan/.condarc
 populated config files : /opt/conda/.condarc
          conda version : 4.5.6
    conda-build version : not installed
         python version : 3.5.2.final.0
       base environment : /opt/conda  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /opt/conda/pkgs
                          /home/jovyan/.conda/pkgs
       envs directories : /opt/conda/envs
                          /home/jovyan/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.6 requests/2.10.0 CPython/3.5.2 Linux/4.9.12-moby debian/8 glibc/2.19
                UID:GID : 1000:100
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.
The command '/bin/sh -c conda create --name snakes python=2' returned a non-zero code: 1
```

### Steps to Reproduce
Running this travis https://github.com/MetaCell/NetPyNE-UI/blob/travis_fix/.travis.yml which in turns builds this Docker file https://github.com/MetaCell/NetPyNE-UI/blob/travis_fix/Dockerfile_dev#L55

**Comment by tarelli** [comment]
Removing from the Docker the `RUN conda update conda` instruction fixes it, so it seems to really point to a Conda update.

```
 matteocantarelli@MacBook-Pro-2  ~/Documents/Development/NetPyNE-UI   travis_fix ● ?  docker build -t="netpyneui04" -f="Dockerfile_dev" .            1 ↵  10589  20:11:19
Sending build context to Docker daemon   805 MB
Step 1/9 : FROM jupyter/base-notebook:eb70bcf1a292
 ---> 3dc2cd3a33c3
Step 2/9 : USER root
 ---> Using cache
 ---> 13814ac8987b
Step 3/9 : ARG netpyneuiBranch=development
 ---> Using cache
 ---> ca515bb004fa
Step 4/9 : ENV netpyneuiBranch ${netpyneuiBranch}
 ---> Using cache
 ---> 809533552ebe
Step 5/9 : RUN echo "$netpyneuiBranch";
 ---> Using cache
 ---> 6253f6091e60
Step 6/9 : RUN apt-get -qq update
 ---> Using cache
 ---> d150cb12d63e
Step 7/9 : RUN apt-get install -y         build-essential         libncurses-dev         python         libpython-dev         cython         libx11-dev         python3-dev         unzip         vim         libpng-dev
 ---> Using cache
 ---> 144c0389f12f
Step 8/9 : USER $NB_USER
 ---> Using cache
 ---> 22f5c9cee244
Step 9/9 : RUN conda create --name snakes python=2
 ---> Running in 3de910393970
Fetching package metadata .........
Solving package specifications: ..........

Package plan for installation in environment /opt/conda/envs/snakes:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2018.4.16  |                0         139 KB  conda-forge
    ncurses-5.9                |               10         1.1 MB  conda-forge
    tk-8.6.7                   |                0         3.1 MB  conda-forge
    zlib-1.2.11                |                0          93 KB  conda-forge
    openssl-1.0.2o             |                0         3.5 MB  conda-forge
    readline-7.0               |                0         838 KB  conda-forge
    sqlite-3.20.1              |                2         1.3 MB  conda-forge
    python-2.7.15              |                0        13.9 MB  conda-forge
    certifi-2018.4.16          |           py27_0         142 KB  conda-forge
    setuptools-39.2.0          |           py27_0         582 KB  conda-forge
    wheel-0.31.1               |           py27_0          61 KB  conda-forge
    pip-9.0.3                  |           py27_0         1.8 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        26.5 MB

The following NEW packages will be INSTALLED:

    ca-certificates: 2018.4.16-0      conda-forge
    certifi:         2018.4.16-py27_0 conda-forge
    ncurses:         5.9-10           conda-forge
    openssl:         1.0.2o-0         conda-forge
    pip:             9.0.3-py27_0     conda-forge
    python:          2.7.15-0         conda-forge
    readline:        7.0-0            conda-forge
    setuptools:      39.2.0-py27_0    conda-forge
    sqlite:          3.20.1-2         conda-forge
    tk:              8.6.7-0          conda-forge
    wheel:           0.31.1-py27_0    conda-forge
    zlib:            1.2.11-0         conda-forge

Proceed ([y]/n)?
Fetching packages ...
ca-certificate 100% |###############################| Time: 0:00:00 444.06 kB/s
ncurses-5.9-10 100% |###############################| Time: 0:00:00   1.73 MB/s
tk-8.6.7-0.tar 100% |###############################| Time: 0:00:00   3.57 MB/s
zlib-1.2.11-0. 100% |###############################| Time: 0:00:00 386.33 kB/s
openssl-1.0.2o 100% |###############################| Time: 0:00:00   3.80 MB/s
readline-7.0-0 100% |###############################| Time: 0:00:00   1.48 MB/s
sqlite-3.20.1- 100% |###############################| Time: 0:00:00   2.01 MB/s
python-2.7.15- 100% |###############################| Time: 0:00:07   1.92 MB/s
certifi-2018.4 100% |###############################| Time: 0:00:00 449.77 kB/s
setuptools-39. 100% |###############################| Time: 0:00:00   1.24 MB/s
wheel-0.31.1-p 100% |###############################| Time: 0:00:00 380.95 kB/s
pip-9.0.3-py27 100% |###############################| Time: 0:00:00   2.52 MB/s
Extracting packages ...
[      COMPLETE      ]|##################################################| 100%
Linking packages ...
[      COMPLETE      ]|##################################################| 100%
#
# To activate this environment, use:
# > source activate snakes
#
# To deactivate this environment, use:
# > source deactivate snakes
#

 ---> 9e9cc032a046
Removing intermediate container 3de910393970
Successfully built 9e9cc032a046
```

**Comment by kalefranz** [comment]
Also unsetting the `NO_PROXY` environment variable will fix it.  The issue is with requests 2.19.  See https://github.com/requests/requests/pull/4723.  Conda 4.5.6 in the `conda-canary` channel has a patch. https://github.com/conda/conda/pull/7498/commits/c88987f9d04e72431d53bfba9e2e4f2da94bd560

**Comment by kalefranz** [comment]
I guess that was 4.5.6 this report is for.  Looking now...

**Comment by kalefranz** [comment]
Ah requests/2.10.0.  Which I guess is probably coming from pip 9.0.3.  I only checked compatibility of that patch back to 2.12.4.  Still looking at a few things...

**Comment by kalefranz** [comment]
Here's why you're falling back to that old `pip`-vendored version of requests

```
jovyan@14516974fa11:~/work$ python
Python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/lib/python3.5/site-packages/requests/__init__.py", line 44, in <module>
    import chardet
ImportError: No module named 'chardet'
```

**Comment by kalefranz** [comment]
That container `jupyter/base-notebook:eb70bcf1a292` starts out with conda 4.2.12.

```
jovyan@791487ca820d:~/work$ conda info
Current conda install:

               platform : linux-64
          conda version : 4.2.12
       conda is private : False
      conda-env version : 4.2.12
    conda-build version : not installed
         python version : 3.5.2.final.0
       requests version : 2.11.1
       root environment : /opt/conda  (writable)
    default environment : /opt/conda
       envs directories : /opt/conda/envs
          package cache : /opt/conda/pkgs
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /opt/conda/.condarc
           offline mode : False
```

That version of conda doesn't know how to properly install noarch python packages, which conda-forge is now using for chardet.

**Comment by kalefranz** [comment]
If you're really stuck on using that ancient container, an upgrade path for you that will work is

```
# Switch to non sudo, create a Python 2 virtual environment 
USER $NB_USER
# Commenting out the conda update things broke!
RUN CONDA_CHANNELS=defaults conda update conda
RUN conda update conda
RUN conda create --name snakes python=2
```

**Comment by tarelli** [comment]
@kalefranz thanks! Hope to migrate to a newer version of the container soon.

**Comment by djhoese** [comment]
I'm having a little trouble following the solutions discussed here, but I'm having the same issue with the vispy travis tests: https://travis-ci.org/vispy/vispy/jobs/402295873#L624

We aren't specifying any old docker containers or anything special and we are downloading the latest miniconda. Anyone have ideas about how I should be getting around this?

As far as I can tell our conda commands come down to:

```bash
$ ./miniconda.sh -b -p ~/anaconda
$ export PATH=~/anaconda/bin:$PATH
$ conda config --add channels conda-forge
$ conda update --yes --quiet conda
# requests got updated to 2.12.4 in the above command, conda to 4.5.6
$ conda create -n testenv --yes --quiet pip python=$PYTHON;
```

**Comment by kalefranz** [comment]
For CI situations, I think the problem goes away when someone from conda-forge merges the PR to release conda 4.5.8.

**Comment by djhoese** [comment]
Pinging @ocefpaf. Is this a known issue ^?

**Comment by kalefranz** [comment]
Solution is https://github.com/conda-forge/conda-feedstock/pull/56

**Comment by Juanlu001** [comment]
@davidmikolas found a similar error with conda 4.5.6:

```
...
File "/Users/david/anaconda/lib/python2.7/site-packages/requests/utils.py", line 617, in get_environ_proxies
        if should_bypass_proxies(url):
    TypeError: should_bypass_proxies_patched() takes exactly 3 arguments (2 given)
```

Should we report it? https://github.com/poliastro/poliastro/issues/405#issuecomment-404074956

**Comment by slzarate** [comment]
Hi, I am also running into the above issue with conda 4.5.6.

**Comment by djhoese** [comment]
The related conda-forge PR has been merged and it seems 4.5.8 is out and fixes this issue.

**Comment by hzitoun** [comment]
Facing same problem with conda v 4.5.6!

**Comment by djhoese** [comment]
@hzitoun The version this is fixed in is 4.5.8. You need to update conda.
